### PR TITLE
[EventEngine] RunAfter migration: Chttp2Connector

### PR DESCRIPTION
**Pull request by yijiem**

### CODE REVIEW

File: src/core/ext/transport/chttp2/client/chttp2_connector.cc [hunk 1]
```diff
@@ -171,13 +174,16 @@ void Chttp2Connector::OnHandshakeDone(void* arg, grpc_error_handle error) {
...
-      GRPC_CLOSURE_INIT(&self->on_timeout_, OnTimeout, self,
-                        grpc_schedule_on_exec_ctx);
-      grpc_timer_init(&self->timer_, self->args_.deadline, &self->on_timeout_);
+      self->timer_handle_ =
+          self->args_.channel_args.GetObject<EventEngine>()->RunAfter(
+              self->args_.deadline - Timestamp::Now(), [self = self->Ref()] {
+                ApplicationCallbackExecCtx callback_exec_ctx;
+                ExecCtx exec_ctx;
+                static_cast<Chttp2Connector*>(self.get())->OnTimeout();
+              });
```
Comment (yijiem) [inline src/core/ext/transport/chttp2/client/chttp2_connector.cc:186]: An alternative is to write something like:

```
// self->Ref() returns a RefCountedPtr<SubchannelConnector>, an
// implicit downcast will be used here:
RefCountedPtr<Chttp2Connector> cc = self->Ref();
self->timer_handle_ =
    self->args_.channel_args.GetObject<EventEngine>()->RunAfter(
        self->args_.deadline - Timestamp::Now(), [self = std::move(cc)] {
          ApplicationCallbackExecCtx callback_exec_ctx;
          ExecCtx exec_ctx;
          self->OnTimeout();
        });
```

I'm not sure which one is more readable or better in other ways. So went with the one without creating an extra `cc` variable.

File: src/core/ext/transport/chttp2/client/chttp2_connector.cc [hunk 2]
```diff
@@ -205,7 +211,15 @@ void Chttp2Connector::OnReceiveSettings(void* arg, grpc_error_handle error) {
         self->result_->Reset();
       }
       self->MaybeNotify(error);
-      grpc_timer_cancel(&self->timer_);
+      if (self->timer_handle_.has_value()) {
+        if (self->args_.channel_args.GetObject<EventEngine>()->Cancel(
```
Comment (ctiller) [inline src/core/ext/transport/chttp2/client/chttp2_connector.cc:215]: We probably don't want to be digging this out of channel args repeatedly

File: src/core/ext/transport/chttp2/client/chttp2_connector.cc (hunk 1 above)
Comment (ctiller) [reply src/core/ext/transport/chttp2/client/chttp2_connector.cc:186]: I like this form

Comment (yijiem) [reply src/core/ext/transport/chttp2/client/chttp2_connector.cc]: Done

File: src/core/ext/transport/chttp2/client/chttp2_connector.cc (hunk 2 above)
Comment (yijiem) [reply src/core/ext/transport/chttp2/client/chttp2_connector.cc]: Added an `event_engine_` member, PTAL.

File: src/core/ext/transport/chttp2/client/chttp2_connector.cc [hunk 3]
```diff
@@ -171,13 +174,17 @@ void Chttp2Connector::OnHandshakeDone(void* arg, grpc_error_handle error) {
...
                         grpc_schedule_on_exec_ctx);
-      self->Ref().release();  // Ref held by OnTimeout()
       grpc_chttp2_transport_start_reading(self->result_->transport,
                                           args->read_buffer,
                                           &self->on_receive_settings_, nullptr);
-      GRPC_CLOSURE_INIT(&self->on_timeout_, OnTimeout, self,
-                        grpc_schedule_on_exec_ctx);
-      grpc_timer_init(&self->timer_, self->args_.deadline, &self->on_timeout_);
+      RefCountedPtr<Chttp2Connector> cc = self->Ref();
+      self->event_engine_ = self->args_.channel_args.GetObject<EventEngine>();
```
Comment (drfloob) [inline src/core/ext/transport/chttp2/client/chttp2_connector.cc:181]: Does it make more sense to store this on `Connect` when the rest of the members are initialized as well?

Comment (yijiem) [reply src/core/ext/transport/chttp2/client/chttp2_connector.cc]: Done, agree it is clearer this way.

**Review by drfloob — APPROVED** [review]

**Pull request merged**